sunpy/coordinates/transformations.py

```diff
@@ -187,71 +221,59 @@ def hcc_to_hgs(helioccoord, heliogframe):
                            "heliographic coordinates for observer '{}' "
                            "without `obstime` being specified.".format(helioccoord.observer))
 
-    x = helioccoord.x.to(u.m)
-    y = helioccoord.y.to(u.m)
-    z = helioccoord.z.to(u.m)
-
-    l0_rad = helioccoord.observer.lon
-    b0_deg = helioccoord.observer.lat
+    # Make sure the observer is in HGS
+    obs = helioccoord.observer.\
+          transform_to(HeliographicStonyhurst(obstime=helioccoord.observer.obstime))
 
-    cosb = np.cos(np.deg2rad(b0_deg))
-    sinb = np.sin(np.deg2rad(b0_deg))
+    total_matrix = _rotation_matrix_hcc_to_hgs(obs.lon, obs.lat)
 
-    hecr = np.sqrt(x**2 + y**2 + z**2)
-    hgln = np.arctan2(x, z * cosb - y * sinb) + l0_rad
-    hglt = np.arcsin((y * cosb + z * sinb) / hecr)
+    # Transform from HCC to HGS at the HCC obstime
+    newrepr = helioccoord.cartesian.transform(total_matrix)
+    int_coord = HeliographicStonyhurst(newrepr, obstime=helioccoord.obstime)
 
-    representation = SphericalRepresentation(
-        np.rad2deg(hgln), np.rad2deg(hglt), hecr.to(u.km))
-    return heliogframe.realize_frame(representation)
+    # Loopback transform HGS if there is a change in obstime
+    if np.any(helioccoord.obstime != heliogframe.obstime):
```

Comment:
Isn't this if statement done in `_transform_obstime`?

Reply:
Yup